Bluetooth: df: Fix sample application build issues

ppryga-nordic opened the pull request:
Fix problem with building sample applications where Host is
split from Controller into separate binaries. In that situation
samples does not have access to controllers Kconfig options.
To allow memory optimization on Host site disable unwanted
features (angle of arrival or angle of departure).

CONFIG_BT_CTLR_DF_CTE_RX is enabled by default if CONFIG_BT_CLTR_DF
is enabled and there is support for DF in given SOC.
If CONFIG_BT_CTLR_DF_SCAN_CTE_RX is disabled then the CONFIG_BT_CTLR-
_PER_SCAN_CTE_NUM_MAX is not available.
It causes build errors if that particular configuration is used by
an application. It was a case for e.g. split builds of direction
finding connectionless TX sample application were periodic advertising
was enabled.

ppryga-nordic committed:
samples: Bluetooth: df: Fix problem with samples split build

Fix problem with building sample applications where Host is
split from Controller into separate binaries. In that situation
samples does not have access to controllers Kconfig options.
To optimize memory usage by an application, disable not needed
features (angle of arrival or angle of departure).

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

ppryga-nordic committed:
Bluetooth: Controller: df: Fix build err if scan cte rx disabled

CONFIG_BT_CTLR_DF_CTE_RX is enabled by default if CONFIG_BT_CLTR_DF
is enabled and there is support for DF in given SOC.
If CONFIG_BT_CTLR_DF_SCAN_CTE_RX is disabled then the CONFIG_BT_CTLR-
_PER_SCAN_CTE_NUM_MAX is not available.
It causes build errors if that particular configuration is used by
an application. It was a case for example in split builds of direction
finding connectionless TX sample application were periodic advertising
was enabled but perddioc advertising sync was disabled.

The commit fixes the problem.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

cvinayak reviewed (changes requested):
Please fix commit title's misspelled `sampels`

ppryga commented:
@cvinayak done

ppryga-nordic committed:
Bluetooth: samples: df: Decrease RAM usage in DF connectionless RX

Direction finding connectionless RX sample didn't fit into
nRF52820 SRAM. The sample will not be able to store IQ samples
for maximum number of CTEs allowed by Bluetooth Core 5.3
specification.

For nRF52820 the new allowed maximum number of scanned CTEs
is set to one, to save more space in SRAM for an application.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

ppryga commented:
Added one commit to fix build issue for nRF53820.

ppryga commented:
@carlescufi @Thalley can you re-review it? I've added a commit to fix another thing 4 days ago (before approval rules were changed) and every approve was discarded.

carlescufi commented:
Please add a 52820 board to the integration platforms

nashif reviewed (dismissed):
you need to update the sample.yaml file to make it build in CI.

ppryga-nordic committed:
samples: Bluetooth: df: add missing integration platforms

Add missing integration plaforms in direction finding samples.
There were no nRF52820 and nRF5340 added.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

ppryga commented:
> Please add a 52820 board to the integration platforms

Added nRF52820 and nRF5340 to sample.yml `integration_platforms`.
